src/Decompiler/Evaluation/ExpressionSimplifier.cs

```diff
@@ -553,6 +559,25 @@ public virtual Expression VisitPhiFunction(PhiFunction pc)
             }
         }
 
+        private Expression SimplifyPhiArg(Expression arg)
```

Comment:
Isn't this is a specialized simplification, where affine expressions of the type `a <binop> b` are being replaced in the phi arguments? Could you not use the `VisitBinaryExpression` method to perform the same simplification? If there is a specific reason why you cannot use `VisitBinaryExpression` you should document it with a comment here.